src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -90,7 +90,7 @@ public static partial class Renamer
                 actions.AddIfNotNull(renameAction);
             }
 
-            if (newDocumentFolders != null && !newDocumentFolders.SequenceEqual(document.Folders))
```

Comment:
Not sure on this, but know what it's trying to solve. Let me think on this change more and see if we can come up with something better